aten/src/ATen/native/cuda/Bucketization.cu

```diff
@@ -106,14 +106,21 @@ void dispatch(Tensor& result, const Tensor& input, const Tensor& boundaries, boo
 
 Tensor& searchsorted_out_cuda(const Tensor& sorted_sequence, const Tensor& self, bool out_int32, bool right, Tensor& result) {
   searchsorted_pre_check(sorted_sequence, self, result, out_int32);
+  Tensor out = result;
```

Comment:
nit: we usually want to declare variables closest to when we need them, could you move this before line 116